modules/internal/src/main/scala/higherkindness/mu/rpc/internal/service/makro/OperationModels.scala

```diff
@@ -0,0 +1,477 @@
+        case Some(ResponseStreaming)      => "SERVER_STREAMING"
+        case Some(BidirectionalStreaming) => "BIDI_STREAMING"
+        case None                         => "UNARY"
+      }
+      q"_root_.io.grpc.MethodDescriptor.MethodType.${TermName(suffix)}"
+    }
+
+    private val updatedName = methodNameStyle match {
+      case Unchanged  => name.toString
+      case Capitalize => name.toString.capitalize
+    }
+
+    private val methodDescriptorName = TermName(s"${updatedName}MethodDescriptor")
+
+    private val methodDescriptorDefName = TermName("methodDescriptor")
+
+    private val methodDescriptorValName = TermName("_methodDescriptor")
+
+    private val reqType           = request.originalType
+    private val reqElemType       = request.messageType
+    private val wrappedRespType   = response.originalType
+    private val unwrappedRespType = response.unwrappedType
+    private val respElemType      = response.messageType
+
+    val methodDescriptorDef: DefDef = q"""
+      def $methodDescriptorDefName(implicit
+        ReqM: _root_.io.grpc.MethodDescriptor.Marshaller[$reqElemType],
+        RespM: _root_.io.grpc.MethodDescriptor.Marshaller[$respElemType]
+      ): _root_.io.grpc.MethodDescriptor[$reqElemType, $respElemType] = {
+        _root_.io.grpc.MethodDescriptor
+          .newBuilder(
+            ReqM,
+            RespM)
+          .setType($streamingMethodType)
+          .setFullMethodName(
+            _root_.io.grpc.MethodDescriptor.generateFullMethodName(
+      ${lit(fullServiceName)}, ${lit(updatedName)}))
+          .build()
+      }
+    """.suppressWarts("Null", "ExplicitImplicitTypes")
+
+    val methodDescriptorVal: ValDef = q"""
+      val $methodDescriptorValName: _root_.io.grpc.MethodDescriptor[$reqElemType, $respElemType] =
+        $methodDescriptorDefName
+    """
+
+    val methodDescriptorObj: ModuleDef = q"""
+      object $methodDescriptorName {
+        $methodDescriptorDef
+        $methodDescriptorVal
+      }
+    """
+
+    private def clientCallMethodFor(clientMethodName: String) =
+      q"$clientCallsImpl.${TermName(clientMethodName)}[$F, $reqElemType, $respElemType](input, $methodDescriptorName.$methodDescriptorValName, channel, options)"
+
+    val clientDef: Tree = streamingType match {
+      case Some(RequestStreaming) =>
+        q"""
+        def $name(input: $reqType): $wrappedRespType =
+          ${clientCallMethodFor("clientStreaming")}
+        """
+      case Some(ResponseStreaming) =>
+        q"""
+        def $name(input: $reqType): $wrappedRespType =
+          ${clientCallMethodFor("serverStreaming")}
+        """
+      case Some(BidirectionalStreaming) =>
+        q"""
+        def $name(input: $reqType): $wrappedRespType =
+          ${clientCallMethodFor("bidiStreaming")}
+        """
+      case None =>
+        q"""
+        def $name(input: $reqType): $wrappedRespType =
+          ${clientCallMethodFor("unary")}
+        """
```

Comment:
Seems like the four options here are identical, except the `clientMethodName`, maybe we can simplify the pattern matching to get that information and leave the rest outside just once?